dhctl/pkg/kubernetes/actions/resources/resource_is_ready.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/kubernetes/client"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/template"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/util/retry"
+)
+
+type resourceIsReadyChecker struct {
+	mutex            sync.Mutex
+	kubeCl           *client.KubernetesClient
+	resourceID       ResourceID
+	watcherIsRunning bool
+	isReady          bool
+}
+
+func newResourceIsReadyChecker(kubeCl *client.KubernetesClient, resource *template.Resource) (Checker, error) {
+	resourceID := ResourceID{
+		Name:             resource.Object.GetName(),
+		Namespace:        resource.Object.GetNamespace(),
+		GroupVersionKind: resource.GVK,
+	}
+
+	checker := &resourceIsReadyChecker{
+		kubeCl:     kubeCl,
+		resourceID: resourceID,
+	}
+
+	return checker, nil
+}
+
+func (c *resourceIsReadyChecker) IsReady() (bool, error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if !c.watcherIsRunning {
+		watcher := NewWatcher(c.kubeCl)
+
+		err := retry.NewSilentLoop(fmt.Sprintf("watch resource '%s'", c.resourceID.String()), 50, 5*time.Second).Run(func() error {
```

Comment:
we have standalone resources creation timeout.
https://github.com/deckhouse/deckhouse/blob/main/dhctl/pkg/kubernetes/actions/resources/resources.go#L206